rend3-anim/src/lib.rs

```diff
@@ -0,0 +1,276 @@
+            for (skin_index, skin) in scene.skins.iter().enumerate() {
+                let skin_index = SkinIndex(skin_index);
+
+                let anim_affected_nodes = &animation_to_joint_nodes[&animation_idx];
+                if skin
+                    .inner
+                    .joints
+                    .iter()
+                    .any(|j| anim_affected_nodes.contains(&NodeIndex(j.inner.node_idx)))
+                {
+                    let entry = animation_skin_usage
+                        .entry(animation_idx)
+                        .or_insert_with(Default::default);
+                    entry.push(skin_index);
+                }
+            }
+        }
+
+        let mut skin_data = FastHashMap::default();
+        for (skin_index, skin) in scene.skins.iter().enumerate() {
+            let skin_index = SkinIndex(skin_index);
+
+            let node_to_joint_idx = skin
+                .inner
+                .joints
+                .iter()
+                .enumerate()
+                .map(|(idx, joint)| (NodeIndex(joint.inner.node_idx), JointIndex(idx)))
+                .collect();
+
+            // Nodes affected by this skin (i.e. joints)
+            let skin_nodes: Vec<NodeIndex> = skin.inner.joints.iter().map(|j| NodeIndex(j.inner.node_idx)).collect();
+
+            let joint_nodes_topological_order: Vec<NodeIndex> = instance
+                .topological_order
+                .iter()
+                .map(|node_idx| NodeIndex(*node_idx))
+                .filter(|node_idx| skin_nodes.contains(node_idx))
+                .collect();
+
+            let skeletons: Vec<SkeletonHandle> = instance
+                .nodes
+                .iter()
+                .flat_map(|node| &node.inner.object)
+                .flat_map(|object| &object.inner.armature)
+                .filter(|armature| armature.skin_index == skin_index.0)
+                .flat_map(|armature| &armature.skeletons)
+                .cloned()
+                .collect();
+
+            skin_data.insert(
+                skin_index,
+                PerSkinData {
+                    node_to_joint_idx,
+                    joint_nodes_topological_order,
+                    skeletons,
+                },
+            );
+        }
+
+        AnimationData {
+            skin_data,
+            animation_skin_usage,
+        }
+    }
+}
+
+/// Helper trait that exposes a generic `lerp` function for various `glam` types
+pub trait Lerp {
+    fn lerp(self, other: Self, t: f32) -> Self;
+}
+impl Lerp for Vec3 {
+    fn lerp(self, other: Self, t: f32) -> Self {
+        self.lerp(other, t)
+    }
+}
+impl Lerp for Quat {
+    fn lerp(self, other: Self, t: f32) -> Self {
+        // Uses Normalized Linear Interpolation (a.k.a. nlerp) as slerp replacement
+        // See: *"Understanding Slerp, Then Not Using It"*
+        // http://number-none.com/product/Understanding%20Slerp,%20Then%20Not%20Using%20It/
```

Comment:
Interesting resource

Reply:
It is! First time I encountered someone just `lerp`ing quaternions I was very confused, but turns out this works just as well in most cases (specifically for animation, I can't notice a difference), as long as you normalize :smile: